### PR TITLE
Documentation for Snapshot 2.0

### PR DESCRIPTION
**Pull request by rivkap**

### CODE REVIEW

File: docs/content/release_notes/limitations.md [hunk 1]
```diff
@@ -80,4 +80,18 @@ The following IBM block storage CSI driver features are not supported on volumes
...
-**Note:** I/O group configuration is only supported for use with IBM Spectrum Virtualize family storage systems.
\ No newline at end of file
+**Note:** I/O group configuration is only supported for use with IBM Spectrum Virtualize family storage systems.
+
+## Snapshot function limitations
+
+**Important:** Snapshot function support is currently only Alpha support.
+
+- Snapshot function is only supported for use with IBM Spectrum Virtualize family storage system versions 8.5.1 or higher. For more information, see **Product overview** > **Technical overview** > **Volume groups** > **Snapshot function** within your Spectrum Virtualize product documentation on [IBM Documentation](https://www.ibm.com/docs).
+- Both source and target PVCs (in a source PVC to snapshot to target PVC scenario) but have the same space efficiency set within their storage classes. If the space efficiency is set differently, the target PVC creation fails.
```
Comment (ArbelNathan) [inline docs/content/release_notes/limitations.md:90]: I dont understand the `but`

File: docs/content/release_notes/limitations.md [hunk 2]
```diff
@@ -80,4 +80,18 @@ The following IBM block storage CSI driver features are not supported on volumes
...
+- A snapshot using the Snapshot function cannot be created with space efficiency set. If the VolumeSnapshotClass has the SpaceEfficiency parameter set along with the snapshot flag enabled, the snapshot creation fails.
+- In very rare cases, there may be leftover or undeleted volumes.
+- A snapshot using the Snapshot function must be created within the same PVC volume or child pool as the original PVC.
```
Comment (ArbelNathan) [inline docs/content/release_notes/limitations.md:93]:
```suggestion
- A snapshot using the Snapshot function must be created within the same PVC pool or child pool as the original PVC.
```?

File: docs/content/configuration/creating_volumesnapshotclass.md [hunk 3]
```diff
@@ -11,6 +11,8 @@ When configuring the file, be sure to use the same array secret and array secret
 -   The `snapshot_name_prefix` parameter is optional.
 
     **Note:** For IBM DS8000® family, the maximum prefix length is five characters.<br/>The maximum prefix length for other systems is 20 characters.<br/>For storage systems that use Spectrum Virtualize, the `CSI` prefix is added as default if not specified by the user.
+
+- The `virt_snap_func` parameter is optional but necessary in Spectrum Virtualize storage systems if using the Snapshot function. To enable the Snapshot function, set the value to _true_. The default value is _false_. If the value is `false` the snapshot will use the FlashCopy function.
```
[the remark this answers is not in the document]
Comment (rivkap) [reply docs/content/configuration/creating_volumesnapshotclass.md:15]: added

File: docs/content/release_notes/limitations.md [hunk 4]
```diff
@@ -80,4 +80,18 @@ The following IBM block storage CSI driver features are not supported on volumes
...
+- Snapshots do not have unique IDs (UUIDs). As a result, if a snapshot with Snapshot function is deleted another snapshot can get the same ID number.
```
[the remark this answers is not in the document]
Comment (rivkap) [reply docs/content/release_notes/limitations.md:94]: thanks - I checked within SV documentation - unique identifier = UID. i'll update accordingly

[the remark this answers is not in the document]
Comment (rivkap) [reply docs/content/release_notes/limitations.md]: done

File: docs/content/release_notes/limitations.md (hunk 2 above)
Comment (rivkap) [reply docs/content/release_notes/limitations.md]: fixed

File: docs/content/release_notes/limitations.md (hunk 1 above)
Comment (rivkap) [reply docs/content/release_notes/limitations.md]: ** typo - must.
 fixed.

File: docs/content/configuration/creating_volumestorageclass.md [hunk 5]
```diff
@@ -57,6 +58,7 @@ Use the `SpaceEfficiency` parameters for each storage system, as defined in the
         io_group: demo-iogrp             # Optional.
         SpaceEfficiency: thin            # Optional.
         volume_name_prefix: demo-prefix  # Optional.
+        virt_snap_func: "false"            # Optional. Values true/false. The default is false.
```
Comment (oriyarde) [inline docs/content/configuration/creating_volumestorageclass.md:61]:
```suggestion
        volume_name_prefix: demo-prefix  # Optional.
        virt_snap_func: "false"          # Optional. Values true/false. The default is false.
```

File: docs/content/release_notes/limitations.md [hunk 6]
```diff
@@ -80,4 +80,18 @@ The following IBM block storage CSI driver features are not supported on volumes
...
\ No newline at end of file
+**Note:** I/O group configuration is only supported for use with IBM Spectrum Virtualize family storage systems.
+
+## Snapshot function limitations
+
+**Important:** Snapshot function support is currently only Alpha support.
+
+- Snapshot function is only supported for use with IBM Spectrum Virtualize family storage system versions 8.5.1 or higher. For more information, see **Product overview** > **Technical overview** > **Volume groups** > **Snapshot function** within your Spectrum Virtualize product documentation on [IBM Documentation](https://www.ibm.com/docs).
+- Both source and target PVCs (in a source PVC to snapshot to target PVC scenario) must have the same space efficiency set within their storage classes. If the space efficiency is set differently, the target PVC creation fails.
+- A snapshot using the Snapshot function cannot be created with space efficiency set. If the VolumeSnapshotClass has the SpaceEfficiency parameter set along with the snapshot flag enabled, the snapshot creation fails.
```
Comment (oriyarde) [inline docs/content/release_notes/limitations.md:91]:
```suggestion
- A snapshot using the Snapshot function cannot be created with space efficiency set. If the `VolumeSnapshotClass` has the `SpaceEfficiency` parameter set along with the `virt_snap_func` flag enabled, the snapshot creation fails.
```

File: docs/content/release_notes/limitations.md [hunk 7]
```diff
@@ -80,4 +80,18 @@ The following IBM block storage CSI driver features are not supported on volumes
...
+- In very rare cases, there may be leftover or undeleted volumes.
+- A snapshot using the Snapshot function must be created within the same PVC or child pool as the original PVC.
```
Comment (oriyarde) [inline docs/content/release_notes/limitations.md:93]:
```suggestion
- A snapshot using the Snapshot function must be created within the same pool or child pool as the original PVC.
```

File: docs/content/release_notes/limitations.md [hunk 8]
```diff
@@ -80,4 +80,18 @@ The following IBM block storage CSI driver features are not supported on volumes
...
+- Snapshots do not have unique identifiers (UIDs). As a result, if a snapshot with Snapshot function is deleted another snapshot can get the same ID number.
+- Any object that is linked in any way (for example, a clone or a snapshot) must have the same definition of snapshot support. For example, a clone cannot be created with FlashCopy mapping enabled from a PVC with an existing Snapshot function connection.
```
Comment (oriyarde) [inline docs/content/release_notes/limitations.md:95]:
```suggestion
- Any object that is linked in any way (for example, a clone or a snapshot) must have the same definition of snapshot support. For example, a clone cannot be created with `virt_snap_func` disabled, from a PVC with an existing Snapshot function connection.
```

File: docs/content/release_notes/limitations.md [hunk 9]
```diff
@@ -80,4 +80,18 @@ The following IBM block storage CSI driver features are not supported on volumes
...
+
+    **Note:** FlashCopy mapping (`fcmap`) and Snapshot function cannot be used together on the same volume however they can be used on different volumes within the same storage system.
```
Comment (oriyarde) [inline docs/content/release_notes/limitations.md:97]:
```suggestion
    **Note:** FlashCopy mapping (`fcmap`) and Snapshot function cannot be used together on the same volume. however, they can be used on different volumes within the same storage system.
```

File: docs/content/release_notes/whats_new.md [hunk 10]
```diff
@@ -4,6 +4,12 @@ IBM® block storage CSI driver 1.10.0 introduces the enhancements that are detai
 
 **General availability date:** 15 July 2022
 
+## Alpha support for the new snapshot function that was introduced in IBM Spectrum Virtualize 8.5.1 release
+
+This versions adds Alpha support for the new snapshot function that was introduced in IBM Spectrum Virtualize 8.5.1 release. The main use case of snapshot is corruption protection. It protects the user data from deliberate or accidental data corruption from the host's systems. For more information about the snapshot function, see **Product overview** > **Technical overview** > **Volume groups** > **Snapshot function** within your Spectrum Virtualize product documentation on [IBM Documentation](https://www.ibm.com/docs).
+
+**Note:** The IBM® FlashCopy and Snapshot function are both referred to as the more generic volume snapshots and cloning within this documentation set. Not all supported products use the FlashCopy adn Snapshot function terminology. Spectrum Virtualize storage systems introduced the new Snapshot function as of Spectrum Virtualize 8.5.1 release. Notes clarifying which function is being referred to within this document are made, as necessary.
```
Comment (oriyarde) [inline docs/content/release_notes/whats_new.md:11]:
```suggestion
**Note:** The IBM® FlashCopy and Snapshot function are both referred to as the more generic volume snapshots and cloning within this documentation set. Not all supported products use the FlashCopy and Snapshot function terminology. Spectrum Virtualize storage systems introduced the new Snapshot function as of Spectrum Virtualize 8.5.1 release. Notes clarifying which function is being referred to within this document are made, as necessary.
```

File: docs/content/release_notes/limitations.md (hunk 7 above)
Comment (rivkap) [reply docs/content/release_notes/limitations.md:93]: @ArbelNathan  please confirm. I already changed per your last comment
image

File: docs/content/release_notes/limitations.md (hunk 6 above)
Comment (rivkap) [reply docs/content/release_notes/limitations.md]: updated

File: docs/content/release_notes/whats_new.md (hunk 10 above)
Comment (rivkap) [reply docs/content/release_notes/whats_new.md]: fixed

File: docs/content/release_notes/limitations.md (hunk 7 above)
Comment (ArbelNathan) [reply docs/content/release_notes/limitations.md]: I approve Ori's change

**Review by ArbelNathan — APPROVED** [review]
LGTM

File: docs/content/release_notes/limitations.md (hunk 7 above)
Comment (rivkap) [reply docs/content/release_notes/limitations.md]: thanks - done

**Review by oriyarde — APPROVED** [review]
LGTM

**Review by ArbelNathan — APPROVED** [review]

File: docs/content/release_notes/changelog_1.10.0.md [hunk 11]
```diff
@@ -1,4 +1,5 @@
 # 1.10.0 (July 2022)
 
 IBM® block storage CSI driver 1.10.0 adds new support and enhancements:
+- Support for Spectrum Virtualize Snapshot function that was introduced in IBM Spectrum Virtualize 8.5.1 release
```
Comment (annaparienti) [inline docs/content/release_notes/changelog_1.10.0.md:4]: isn't it supposed to be 8.5.1.0?

Comment (rivkap) [reply docs/content/release_notes/changelog_1.10.0.md:4]: no. 8.5.1 - per SV documentation as well
image

**Review by annaparienti — APPROVED** [review]
LGTM

**Comment by samishekh** [comment]
LGTM

**Review by samishekh — APPROVED** [review]
LGTM

**Pull request merged**